torch/fx/proxy.py

```diff
@@ -370,6 +370,9 @@ class Proxy:
     For a more detailed description into the Proxy internals, check out
     the "Proxy" section in `torch/fx/OVERVIEW.md`
     """
+    # Indicates to our torch_function dispatching infra that
+    # this is an "infra" object
+    _mode_key = torch._C._TorchFunctionModeKey.PROXY
```

Comment:
After the short-term fix, I don't actually have a test case that verifies this is doing something...